pkg/querier/queryrange/roundtrip_test.go

```diff
@@ -856,19 +855,13 @@ func TestNewTripperware_Caches(t *testing.T) {
 				if s != nil {
 					c, ok := s.(cache.Cache)
 					require.True(t, ok)
+
+					require.NotNil(t, c)
 					caches = append(caches, c)
 				}
 			}
 
 			require.Equal(t, tc.numCaches, len(caches))
-
-			if tc.numCaches == 2 {
-				if tc.equalCaches {
-					require.Equal(t, caches[0], caches[1])
```

Comment:
cannot compare them as they are bound to be different instances of embedded cache.
what we want to compare are configs passed down to the cache

> results cache enabled, stats cache enabled but different
I'll raise a separate pr to improve this test